### PR TITLE
py-flake8: bump including deps

### PR DESCRIPTION
**Pull request by haampie**
So we can get the same new pedantic rules in Spack :laughing:

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-pycodestyle/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pycodestyle/package.py:1]: Missing a dep:
```python
depends_on("python@3.8:", when="@2.11:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pyflakes/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyflakes/package.py:1]: Missing a dep:
```python
depends_on("python@3.8:", when="@3.1:", type=("build", "run"))
```

**Review by adamjstewart — COMMENTED** [review]
@haampie it's better to wait until package maintainers have a chance to review before merging.

**Pull request merged**
